### ISSUE TITLE
[BUG][Codegen] 'Required property not in properties' error when the second referenced model is a combined model

### ISSUE THREAD

**Issue by dorothea-kalliora** [OpenAPITools/openapi-generator#17863]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

From version `7.3.0` when referencing two models using the `oneOf` keyword and:
1. the first model has a required and **defined** property named `property1` and
2. the second model is a combined model using the `allOf` keyword

we get this unexpected error in the logs:
```
[main] ERROR o.o.codegen.DefaultCodegen - Required var property1 not in properties
[main] ERROR o.o.codegen.DefaultCodegen - Required var property1 not in properties
```

Although we get this error, the generator doesn't exit and generates the api code.

We also see this error in the logs, when building the project from master following these https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator instructions. Here is one of the logs:
```
...
[TestNG-test=Surefire test-3] ERROR org.openapitools.codegen.DefaultCodegen - Required var seeds not in properties
...
```
 
##### openapi-generator version

version `7.3.0`

##### OpenAPI declaration file content or url

Here is the minimal `yaml` file that produces the above errors:

```yaml
openapi: 3.0.3
info:
  title: Test
  version: 1.0.0
paths:
  /api/v1/user:
    get:
      operationId: getUser
      responses: 
        '200':
          description: Successfully fetched information for the user
          content:
            application/json:
              schema:
                oneOf:
                  - $ref: '#/components/schemas/UserInfoPhone'
                  - $ref: '#/components/schemas/UserInfoEmail'
        '401':
          description: Failed to authenticate request
components:
  schemas:
    User:
      type: object
      properties:
        name:
          type: string
        surname:
           type: string
    UserInfoEmail:
      allOf:
        - $ref: '#/components/schemas/User'
        - type: object
          properties:
            email:
              type: string
    UserInfoPhone:
        type: object
        required:
          - phone
        properties:
          phone:
            type: string
```

##### Steps to reproduce

To reproduce the errors, you can run:
```
openapi-generator-cli generate -i test.yaml -g typescript-axios
```

where `test.yaml` contains the above script.

The errors we get when generating the code are:
```
[main] ERROR o.o.codegen.DefaultCodegen - Required var phone not in properties
[main] ERROR o.o.codegen.DefaultCodegen - Required var phone not in properties
```

##### Related issues/PRs

No related PRs/issues found.

##### Suggest a fix

The above error log is introduced by this commit https://github.com/OpenAPITools/openapi-generator/commit/64f2cad9e8995db1d92c718576be3f5ca49da02e.

We noticed that we don't get the error if we use one of the following options:

- we reverse the order we reference the components in the `oneOf` keyword
- we don't use the `allOf` keyword to extend a model (e.g. in the above example define the properties of the `User` model directly on the `UserInfoEmail` model)
- we use the `--openapi-normalizer REF_AS_PARENT_IN_ALLOF=true` argument when generating the code.

**Comment by MattTheRed** [comment]
Wondering if there has been any progress on this or if there is a recommended workaround

**Comment by fcarrero** [comment]
same with this issue.

**Comment by wfajkowski** [comment]
bump

**Comment by mdeheus** [comment]
We have the same issue. We consider rolling back to openapi generator version 7.2.0 to avoid the 20+ errors in the log during compilation.

**Comment by blouflashdb** [comment]
Same Problem here with Version 7.11

**Comment by wing328** [comment]
thanks for reporting the issue.

One way to use `debug` instead of `error` assuming there's actually no change in the output. In other words, the error message is false alarm.

Can you guys please confirm there's no change in the output when upgrading from v7.2.0 to newer versions?


cc @martin-mfg (author of https://github.com/OpenAPITools/openapi-generator/commit/64f2cad9e8995db1d92c718576be3f5ca49da02e)

**Comment by martin-mfg** [comment]
**TL;DR:** It's more or less a false alarm and I will remove it.

As pointed out in this thread, the error message was introduced in https://github.com/OpenAPITools/openapi-generator/commit/64f2cad9e8995db1d92c718576be3f5ca49da02e, via a PR opened by me. This change only added the printed _message_, the reported "problem" itself existed already before. And the reported "problem" is in fact not (always) a problem. At the specific point in the code, it can be totally fine that a property is not part of the `properties` collection.

On the other hand, in certain situations it might in fact be a problem. OpenAPI Generator for sure has shortcomings when processing allOf/anyOf/oneOf specs. E.g. generating Java code from [modules/openapi-generator/src/test/resources/3_0/anyOfDiscriminator.yaml](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/3_0/anyOfDiscriminator.yaml) triggers the discussed error message - and also it results in code which doesn't compile and which misses some of the specified properties. But this can be addressed in separate PRs by other contributors.

Lastly, the discussed error message was part of my PR only because I [accidentally included](https://github.com/OpenAPITools/openapi-generator/pull/16976#issuecomment-2026939160) changes by @dabdirb from [his PR](https://github.com/OpenAPITools/openapi-generator/pull/16976) in my PR. So maybe @dabdirb has further insights about this error message?
